mono/mini/Makefile.am.in

```diff
@@ -826,7 +826,11 @@ version.h: Makefile
 	if test -d $(top_srcdir)/.git; then \
 		(cd $(top_srcdir); \
 			LANG=C; export LANG; \
-			branch=`git branch | grep '^\*' | sed 's/(detached from .*/explicit/' | cut -d ' ' -f 2`; \
+			if test -z "$$ghprbPullId"; then \
+				branch=`git branch | grep '^\*' | sed 's/(detached from .*/explicit/' | cut -d ' ' -f 2`; \
+			else \
+				branch="$$ghprbPullId"; \
```

Comment:
I'd prefer something like "Pull Request #3015" or "pull-request-3015" here, otherwise we'll end up with 3015/ab3fe8a01 which is less clear.